pkg/phases/argocdoperator/test.go

```diff
@@ -0,0 +1,88 @@
+package argocdoperator
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/flanksource/commons/console"
+	argo "github.com/flanksource/karina/pkg/api/argocd"
+	"github.com/flanksource/karina/pkg/platform"
+	"github.com/flanksource/kommons"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func NewArgoCDClusterConfig(name string) *argo.ArgoCD {
+	return &argo.ArgoCD{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ArgoCD",
+			APIVersion: "argoproj.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+}
+
+func Test(p *platform.Platform, test *console.TestResults) {
+	if p.ArgoCDOperator.IsDisabled() {
+		test.Skipf("argocd-operator", "ArgoCD Operator is disabled")
+		return
+	}
+	client, _ := p.GetClientset()
+	kommons.TestNamespace(client, Namespace, test)
+	if p.E2E {
+		TestE2E(p, test)
+	}
+}
+
+func TestE2E(p *platform.Platform, test *console.TestResults) {
+	testName := "argocd-operator-e2e"
+	if p.ArgoCDOperator.IsDisabled() {
```

Comment:
I don't think this will ever be true ?

Reply:
So if the user defines `argocdOperator.disabled: true` or they didn't define `argocdOperator.version` then this will be true.

Reply:
correct